sky/provision/gcp/instance.py

```diff
@@ -228,10 +218,14 @@ def cleanup_ports(
 ) -> None:
     """See sky/provision/__init__.py"""
     assert provider_config is not None, cluster_name_on_cloud
-    if 'firewall_rule' not in provider_config:
-        # No new ports were opened, so there is nothing to clean up.
-        return
     project_id = provider_config['project_id']
-    firewall_rule_name = provider_config['firewall_rule']
-    instance_utils.GCPComputeInstance.delete_firewall_rule(
-        project_id, firewall_rule_name)
+    if 'ports' in provider_config:
+        # Backward compatibility for old provider config.
```

Comment:
Add todo for when to remove it, e.g. after 0.6.0.

Reply:
Done. Thanks!